inc/db_pgsql.php

```diff
@@ -1271,9 +1271,13 @@ function drop_table($table, $hard=false, $table_prefix=true)
 			$table_prefix = $this->table_prefix;
 		}
 
+		$table_prefix_bak = $this->table_prefix;
+		$this->table_prefix = '';
```

Comment:
What's the reason for doing this? I see on line 1290 that you assign it `$this->table_prefix = $table_prefix_bak` without accessing it at all inbetween line 1274 to 1290? You don't change `$table_prefix_bak` either after assigning `$this->table_prefix` to it.

You can remove this line, and then you can remove line 1290, at least it seems very redundant, maybe I am wrong, what's the reasoning?

Reply:
`table_exists()` uses `$this->table_prefix`, but yeah the code looks a bit redundant and confusing.

Reply:
It indeed does use `$this->table_prefix` but it does not change the value of it.

Reply:
The function `drop_table()` permits dropping any tables with any prefix. So you could write this part as either:
```php
$table_prefix_bak = $this->table_prefix;
$this->table_prefix = '';
// ...
$this->table_exists($table_prefix.$table);
// ...
$this->table_prefix = $table_prefix_bak;
```
or
```php
$table_prefix_bak = $this->table_prefix;
$this->table_prefix = $table_prefix;
// ...
$this->table_exists($table);
// ...
$this->table_prefix = $table_prefix_bak;
```

Reply:
What stops you from doing
```
$this->table_exists($table_prefix.$table);
```
Without any need to store `$this->table_prefix` in a temporary variable?

Reply:
Any non-empty `$table_prefix`.

Reply:
@Xazin that means:

 - `table_exists($table_prefix.$table)` will check the existence of table of name `$table_prefix.$this->table_prefix.$table`, so the first way sets `$this->table_prefix` to `''`.
 - `table_exists($table)` will check the table of name `$this->table_prefix.$table`, so the second way sets `$this->table_prefix` to `$table_prefix`.

and both of them save the original `$this->table_prefix` as `$table_prefix_bak` and restore it afterwards.

I don't know what confuses you exactly, or what is the way you'd like to do about this part?

Reply:
Other solutions that may be acceptable:
- add similar `bool $table_prefix` support to `DB_Base::table_exists()`
- remove `$table_prefix` support altogether (appears to be only used in `upgrade1.php`)